apps/chat/src/components/Chat/ModelsDialog.tsx

```diff
@@ -78,12 +95,22 @@ export const ModelsDialog: FC<ModelsDialogProps> = ({
         (entity) => entity.type === EntityType.Assistant,
       ),
     );
-    setFilteredApplicationsEntities(
-      newFilteredEntities.filter(
-        (entity) => entity.type === EntityType.Application,
-      ),
-    );
-  }, [models, entityTypes, searchTerm]);
+    if (isCustomApplicationsEnabled) {
+      setFilteredApplicationsEntities(
+        newFilteredEntities.filter(
+          (entity) => entity.type === EntityType.Application,
+        ),
+      );
+    } else {
+      setFilteredApplicationsEntities(
```

Comment:
Maybe we need to discuss it on the daily call? Should we hide created applications if flag was disabled?